collector/netdev_common.go

```diff
@@ -46,31 +49,43 @@ func init() {
 
 // NewNetDevCollector returns a new Collector exposing network device stats.
 func NewNetDevCollector(logger log.Logger) (Collector, error) {
-	if *netdevIgnoredDevices != "" && *netdevAcceptDevices != "" {
-		return nil, errors.New("device-blacklist & accept-devices are mutually exclusive")
+	if *oldNetdevDeviceInclude != "" {
+		level.Warn(logger).Log("msg", "--collector.netdev.device-whitelist is DEPRECATED, use --collector.netdev.device-include")
+		*netdevDeviceInclude = *oldNetdevDeviceInclude
 	}
 
-	var ignorePattern *regexp.Regexp
-	if *netdevIgnoredDevices != "" {
-		ignorePattern = regexp.MustCompile(*netdevIgnoredDevices)
+	if *oldNetdevDeviceExclude != "" {
+		level.Warn(logger).Log("msg", "--collector.netdev.device-blacklist is DEPRECATED, use --collector.netdev.device-exclude")
+		*netdevDeviceExclude = *oldNetdevDeviceExclude
```

Comment:
What if both were set?

Reply:
If both are set, the new flag is ignored. I am attempting to keep this backwards compatibility code as simple as possible.

Reply:
The code is simple but the behavior on end users is not really obvious.

Reply:
I agree. What logic flow should we follow?

* Error if both are specified?
* Use new flag and ignore old flag?

I'm open to ideas here.

Reply:
Error to make sure the user does not think they add up.